contracts/Core/BlockManager.sol

```diff
@@ -7,38 +7,21 @@ import "./interface/IRewardManager.sol";
 import "./interface/IVoteManager.sol";
 import "./interface/IAssetManager.sol";
 import "./storage/BlockStorage.sol";
+import "./StateManager.sol";
 import "../lib/Random.sol";
 import "../Initializable.sol";
 import "./ACL.sol";
 
-contract BlockManager is Initializable, ACL, BlockStorage {
+contract BlockManager is Initializable, ACL, BlockStorage, StateManager {
     IParameters public parameters;
     IStakeManager public stakeManager;
     IRewardManager public rewardManager;
     IVoteManager public voteManager;
     IAssetManager public assetManager;
 
-    event BlockConfirmed(uint256 epoch, uint256 stakerId, uint256[] medians, uint256[] ids, uint256 timestamp);
+    event BlockConfirmed(uint32 epoch, uint32 stakerId, uint32[] medians, uint256 timestamp);
```

Comment:
I think we should change values32 to values48 as we will face an error of value out of bounds with datatype _uint32_. I tried setting the value of ETH that is being reported by stakers on razorscan to a variable with datatype _uint32_ and this was the following error

![Screenshot from 2021-08-09 15-25-10](https://user-images.githubusercontent.com/52957842/128690440-9e8b5414-de7c-4761-a1b5-fc19457bd34a.png)

Value being reported at the time of writing this comment:- 302968000000
uint48 seems to be the safest and efficient datatype for our requirements

Reply:
this is why https://github.com/razor-network/contracts/issues/266 is created
we will set a multiplier of lets say 100. then stakes will report value 302968.
i can increase it for now. and reduce it to uint32 later when #266 is solved.

Reply:
I think its better if it is increased for now and reduced it when #266 is solved

Reply:
done